nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm

```diff
@@ -0,0 +1,107 @@
+	bound_width = 32
+	pixel_x = 0
+	pixel_y = 0
+	idle_power_usage =  2000
+	var/obj/machinery/ship_weapon/energy/parent
+	var/on = FALSE
+	density = TRUE
+	critical_machine = TRUE
+
+/obj/machinery/cooling/cooler
+	name = "subspace cooling unit"
+	desc = "A cooling unit that dumps the massive amounts of heat energy weapons generate into subspace."
+	circuit = /obj/item/circuitboard/machine/cooling
+
+
+
+/obj/machinery/cooling/cooler/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src) //I have no idea what I'm doing and this causes errors so
+	E.coolers |= src
+	parent = E
+
+/obj/machinery/cooling/storage/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src)
+	E.storages |= src
+	parent = E
+
+
+/obj/machinery/cooling/cooler/Destroy()
+  parent.coolers -= src
+  . = ..()
+
+/obj/machinery/cooling/storage/Destroy()
+  parent.storages -= src
+  . = ..()
+
+/obj/machinery/cooling/attack_hand(mob/user)
+	. = ..()
+	if(panel_open)
+		to_chat(user, "<span class='notice'>You must turn close the panel on [src] before turning it on.</span>")
+		return
+	to_chat(user, "<span class='notice'>You press [src]'s power button.</span>")
+	on = !on
+	update_icon()
+
+/obj/machinery/cooling/examine()
+	. = ..()
+	if(on)
+		. += "The power is on"
+	if(parent)
+		. += "The thermal subspace transcever is linked"
+	else
+		. += "it's completely inactive"
+
+
+/obj/machinery/cooling/update_icon()
+	cut_overlays()
+	if(panel_open)
+		icon_state = "smes-o"
+	if(on && parent)
+		add_overlay("smes-op1")
+	if(on)
+		add_overlay("smes-oc1")
+	else
+		add_overlay("smes-op0")
+
+
+/obj/machinery/cooling/multitool_act(mob/living/user, obj/item/multitool/I)
+	if(!multitool_check_buffer(user, I))
+		return
+	var/obj/item/multitool/P = I
+
+	if(istype(P.buffer, /obj/machinery/ship_weapon/energy))
+		if(get_area(P.buffer) != get_area(src))
+			to_chat(user, "<font color = #666633>-% Cannot link machines across power zones. %-</font color>")
+			return
+
+		parent = P.buffer
+		if(istype(src,/obj/machinery/cooling/cooler))
+			.=TRUE
+			parent.coolers |= src
+			to_chat(user, "<font color = #666633>-% Successfully linked [P.buffer] with [src] %-</font color>")
+		if(istype(src,/obj/machinery/cooling/storage))
+			.=TRUE
+			parent.storages |= src
+			to_chat(user, "<font color = #666633>-% Successfully linked [P.buffer] with [src] %-</font color>")
+		if(length(parent.storages)+length(parent.coolers) >= 10)
+			var/E = pick(parent.storages+parent.coolers)
+			explosion(get_turf(E), 0, 1, 3, 5, flame_range = 4)
+			return
+
+		return
+
+
+/obj/machinery/cooling/storage
+	name = "subspace heatsink unit"
+	desc = "A cooling unit that stores the massive amounts of heat energy weapons generate in subspace."
+	icon_state = "smes"
+	circuit = /obj/item/circuitboard/machine/cooling/storage
+
```

Comment:
I'm going to be honest, I don't like how most of this file and code is written, would you like me to rewrite it or do you prefer that I just give you tips?
If you do not have any sprites yet, I would be happy to make some. Even if you don't, it is usually appreciated to have *some* distinguishing features between your own machines and already existing ones. Even if it's only a red and blue stripe, just those things can be enough for telling apart two components.

Reply:
I have plans to put together sprites, but I haven't taken any steps to do so, if you want to add some sprites, either placeholder or fully completed, that wouldn't be bad at all